evaluators/xpath-evaluator/components/org.wso2.carbon.gateway.evaluators.xpathevaluator/src/main/java/org/wso2/carbon/gateway/evaluators/xpathevaluator/XPathEvaluator.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ * <p>
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.gateway.evaluators.xpathevaluator;
+
+import net.sf.saxon.s9api.DocumentBuilder;
+import net.sf.saxon.s9api.Processor;
+import net.sf.saxon.s9api.SaxonApiException;
+import net.sf.saxon.s9api.XPathCompiler;
+import net.sf.saxon.s9api.XPathSelector;
+import net.sf.saxon.s9api.XdmNode;
+import org.apache.axiom.om.OMElement;
+import org.wso2.carbon.gateway.core.Constants;
+import org.wso2.carbon.gateway.core.flow.contentaware.abstractcontext.MessageBodyEvaluator;
+import org.wso2.carbon.gateway.core.flow.contentaware.exceptions.MessageBodyEvaluationException;
+
+import java.io.StringReader;
+import javax.xml.transform.stream.StreamSource;
+
+/**
+ * This class evaluates xpath expressions.
+ */
+public class XPathEvaluator implements MessageBodyEvaluator {
+    private XPathCompiler xPathCompiler;
```

Comment:
I like to suggest, let's document these variables religiously (explaining why it exists, responsibility etc). rationale: for at least next 5-10 years we will end up reading what we (and other folks - new devs, external users, clients ) write today with varying urgencies ( L1s , L2s, war rooms, future improvements etc). having these documented adds value to code.